arch/x86/adler32_avx512.c

```diff
@@ -71,7 +78,14 @@ static inline uint32_t adler32_fold_copy_impl(uint32_t adler, uint8_t *dst, cons
             k -= 64;
 
             __m512i vs1_sad = _mm512_sad_epu8(vbuf, zero);
+#if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable: 4310 )
+#endif
             __m512i v_short_sum2 = _mm512_maddubs_epi16(vbuf, dot2v);
```

Comment:
```c
__m512i v_short_sum2 = _mm512_castps_si512(_mm512_maddubs_epi16(vbuf, dot2v));
```
instead?

Reply:
Not really familiar with AVX512 intrinsics, but changed...

Reply:
Yeah I'm not entirely sure what you're trying to do with that cast.  Nothing here should be interpreted as floating point.

Reply:
Internally it used the intrinsic version with mask, but the mask was incorrect at least up to v141 toolchain. I undefined and redefined with correct value for the mask.

Reply:
Heh, that's odd.  My question was more directed at @nmoinvaz though, that cast is for single precision floats, the compiler is definitely not going to like that.  It is silly that we have to do this at all for MSVC, though.  For one, always emitting the masked variant of that instruction means that you always have to source a k register to do that.  Which means you're probably shuffling k registers in and out of GPRs when you actually have contention (assuming their optimizer doesn't try to optimize out a constant mask, anyway).

Reply:
@KungFuJesus Wouldn't have been surprise if Microsoft messes up return type of some intrinsics as they're pretty much using only two types instead of having element bit width and element count in the type itself.

I can't just assume they managed to get something correct in one SDK version... I've been working with Microsoft since 1998 and I've noticed some of their answers are quite hilarious suggesting most of the time they have used outsourced/third-party code and just hacked their own branding.